File: tv/src/main/java/com/synergics/stb/iptv/leanback/etc/Const.java
package com.synergics.stb.iptv.leanback.etc;

import com.synergics.stb.iptv.leanback.BaseApplication;
import com.synergics.stb.iptv.leanback.models.TVCategories;

import java.util.List;

public class Const {
    public static void firstRun(boolean firstrun){
        BaseApplication.sp.edit().putBoolean("first-run", firstrun).apply();
    }

    public static boolean firstRun(){
        return BaseApplication.sp.getBoolean("first-run", false);
    }

    public static TVCategories search(List<TVCategories> list, String filter){
        TVCategories output = null;
        for (TVCategories tv: list) {
            if (tv.getNama() != null && tv.getNama().equals(filter))
                output = tv;
        }

        return output;
    }
}

File: tv/src/main/java/com/synergics/stb/iptv/leanback/etc/Const.kt
package com.synergics.stb.iptv.leanback.etc

import com.synergics.stb.iptv.leanback.BaseApplication
import com.synergics.stb.iptv.leanback.models.TVCategories

object Const {
    fun firstRun(firstrun: Boolean) {
        BaseApplication.Companion.sp!!.edit().putBoolean("first-run", firstrun).apply()
    }

    fun firstRun(): Boolean {
        return BaseApplication.Companion.sp!!.getBoolean("first-run", false)
    }

    fun search(list: List<TVCategories>, filter: String): TVCategories? {
        var output: TVCategories? = null
        for (tv in list) {
            if (tv.nama != null && tv.nama == filter) output = tv
        }
        return output
    }
}